### ISSUE TITLE
mWeb/Safari - Attachments - New tab opens when clicking the download button on a .doc file

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#43857]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.84-2
**Reproducible in staging?:** Y 
**Reproducible in production?:** Y 
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/4636547
**Email or phone of affected tester (no customers):** dave0123seife+admin10@gmail.com
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Navigate to a conversation
2. Upload a .doc attachment to the chat
3. Click on the download icon found on the .doc file

## Expected Result:
Download starts and no new tab is opened

## Actual Result:
A new tab opens and the download starts

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/84e6629d-4d7c-4da1-8bfd-61166e16cd56

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by lanitochka17** [comment]
@jliexpensify  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by lanitochka17** [comment]
We think that this bug might be related to #vip-vsp

**Comment by jliexpensify** [comment]
ios <> Android swapsies

**Comment by nkdengineer** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
A new tab opens and the download starts

### What is the root cause of that problem?
We pass `shouldOpenExternalLink` as `true` if the browser is safari. It's added to fix this issue https://github.com/Expensify/App/pull/37131.

https://github.com/Expensify/App/blob/b33542383ac2cb8ea1e922aa12dd2265efccd0ab/src/components/AnchorForAttachmentsOnly/BaseAnchorForAttachmentsOnly.tsx#L52


### What changes do you think we should make in order to solve the problem?
I think we don't need to open a new tab when downloading the file in mWeb safari. 

```
fileDownload(sourceURLWithAuth, displayName, '').then(() => Download.setDownload(sourceID, false)); 
```
https://github.com/Expensify/App/blob/b33542383ac2cb8ea1e922aa12dd2265efccd0ab/src/components/AnchorForAttachmentsOnly/BaseAnchorForAttachmentsOnly.tsx#L52

Instead, we will do nothing when clicking on the download button if a file is downloading in mWeb safari. To do that we can get all download collection and check we're downloading a file or not. Or we can create another Onyx key to know whether we download a file.
### What alternative solutions did you explore? (Optional)

**Comment by DylanDylann** [comment]
@Krishna2323 Please keep your proposal consistent.

> Please re-state the problem that we are trying to solve in this issue.
you said: "new tab open" is the problem that we are trying to solve in this issue
 
> What changes do you think we should make in order to solve the problem?
> you suggest: new tab should open when clicking download button on context menu"
  
Two your sections are conflict

**Comment by DylanDylann** [comment]
@Krishna2323 One more thing

> We should also check on other places where fileDownload is used.

Which places you want to mention here?

**Comment by DylanDylann** [comment]
> mWeb/Safari - Attachments - New tab opens when clicking the download button on a .doc file

This is not the problem you are trying to solve

**Comment by DylanDylann** [comment]
In this section

> Please re-state the problem that we are trying to solve in this issue.

Please re-state the problem you see and you are trying to resolve in your proposal. I don't see any value in this section when you copy the issue title. On the contrary, it confuses me

**Comment by nkdengineer** [comment]
@DylanDylann The test branch [here](https://github.com/nkdengineer/App/tree/fix/43857).

**Comment by DylanDylann** [comment]
@dukenv0307 will take this issue

**Comment by kadiealexander** [comment]
@dukenv0307 sorry for the delay! Please take a look at the proposals.

**Comment by srikarparsi** [comment]
Yes, this sounds good. I updated the issue's title and description.

**Comment by kadiealexander** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@dukenv0307] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@dukenv0307] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@dukenv0307] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@dukenv0307] Determine if we should create a regression test for this bug.
- [ ] [@dukenv0307] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@kadiealexander] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by kadiealexander** [comment]
Payouts due:

- [x] Contributor: $250 @Krishna2323 ([here](https://www.upwork.com/nx/wm/workroom/37713989/overview))
- [ ] Contributor+: $250 @dukenv0307 ([here](https://www.upwork.com/nx/wm/workroom/37743796/overview))

Upwork job is [here](https://www.upwork.com/jobs/~018590692829acb48f).
